src/PostgREST/Query.hs

```diff
@@ -252,6 +254,13 @@ failExceedsMaxAffectedPref (Just (PreferMaxAffected n), handling) RSStandard{rsQ
   lift SQL.condemn
   throwError $ Error.ApiRequestError . Error.MaxAffectedViolationError $ toInteger queryTotal
 
+-- | Fail max-affected pref when used with handling=strict but rpc returns scalar
+failMaxAffectedRpcReturnsSingle :: (Maybe PreferMaxAffected, Maybe PreferHandling) -> Routine -> DbHandler ()
+failMaxAffectedRpcReturnsSingle (Nothing,_) _ = pure ()
+failMaxAffectedRpcReturnsSingle (Just (PreferMaxAffected _), handling) rout = when (funcReturnsSingle rout && handling == Just Strict) $ do
+    lift SQL.condemn
```

Comment:
Q: It looks like this will start an empty transaction?

Since this failure only depends on the schema cache + header, there would be no need to start a transaction right?

Reply:
So it seems to me that this failure should be produced at the `Plan` module, not at the `Query` one.